engine/lib/phases/phase_drop_match_guards.ml

```diff
@@ -60,12 +60,19 @@ module%inlined_contents Make (F : Features.T) = struct
 
     [%%inline_defs dmutability]
 
+    let maybe_simplified_match scrutinee (arms : B.arm list) =
+      match arms with
+      (* If there is only one wildcard branch we can simplify *)
+      | [ { arm = { body; arm_pat = { p = PWild; _ }; _ }; _ } ] -> body.e
+      (* General case *)
+      | _ -> Match { scrutinee; arms }
+
     let rec dexpr' (span : span) (expr : A.expr') : B.expr' =
       match expr with
       | [%inline_arms "dexpr'.*" - Match] -> auto
       | Match { scrutinee; arms } ->
           let arms = transform_arms (dexpr scrutinee) (List.rev arms) [] in
-          Match { scrutinee = dexpr scrutinee; arms }
+          maybe_simplified_match (dexpr scrutinee) arms
```

Comment:
Sorry for being picky :grin: but can do `maybe_simplified_match` iff the user did not type a match with one arm with `_` LHS and no guard? otherwise such user typed matches are optimized away, e.g. https://hax-playground.cryspen.com/#fstar/365ad6f8bf/gist=be0c4a2ebe27079803f981fc39c45283

Reply:
No problem, I fixed that now!